android/java/org/chromium/chrome/browser/tab/BraveTab.java

```diff
@@ -0,0 +1,79 @@
+
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.tab;
+
+import android.content.SharedPreferences;
+import java.util.Calendar;
+import android.os.Build;
+
+import androidx.annotation.Nullable;
+
+import org.chromium.base.ContextUtils;
+import org.chromium.chrome.browser.tab.Tab;
+import org.chromium.chrome.browser.tabmodel.TabLaunchType;
+import org.chromium.chrome.browser.preferences.BackgroundImagesPreferences;
+import org.chromium.chrome.browser.ntp.sponsored.BackgroundImage;
+import org.chromium.chrome.browser.ntp.sponsored.SponsoredImage;
+import org.chromium.chrome.browser.ntp.sponsored.SponsoredImageUtil;
+
+public class BraveTab extends Tab {
+	private static final String TAG = "BraveTab";
+
+	private BackgroundImage backgroundImage;
+    private int index;
+
+	BraveTab(int id, Tab parent, boolean incognito, @Nullable @TabLaunchType Integer launchType) {
+		super(id,parent,incognito,launchType);
+
+		if (Build.VERSION.SDK_INT > Build.VERSION_CODES.M) {
+            backgroundImage = getBackgroundImage();
+        }
+        index = SponsoredImageUtil.imageIndex;
+	}
+
+	public BackgroundImage getTabBackgroundImage() {
+        return backgroundImage;
+    }
+
+    public int getIndex() {
+        return index;
+    }
+
+    private BackgroundImage getBackgroundImage() {
+        BackgroundImage backgroundImage;
+        SharedPreferences mSharedPreferences = ContextUtils.getAppSharedPreferences();
+
+        if (mSharedPreferences.getInt(BackgroundImagesPreferences.PREF_APP_OPEN_COUNT, 0) == 2
+            && SponsoredImageUtil.imageIndex == 2) {
```

Comment:
Seems like this logic would be better encapsulated inside `SponsoredImageUtil` and exposing public instance variables is generally considered to be an OOP anti-pattern